### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by theopak**
~~_Please do not merge yet._ I'm actively working on this PR, and opening it now so that Travis-CI + Coveralls checks everything correctly.~~ UPDATE: I'm done. Looks ok?

**TODO: Version 1.5.9**
- [x] Replace all instances of concatenation with `path.sep` with use of `.path.join()` or `path.resolve()` instead.
- [x] Ignore non-config files in config directory
- [x] Add test cases for having dot files in config directory.
- [x] Add test cases for having non-json, non-xml files in config directory.
- [x] Update doc in README
- [x] Use snake_case_because_it_matches_existing_code

### CODE REVIEW

File: utils/keygen.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ exports.createKey = function(root_dir, from_port, to_port, key_id, secret, cb) {
     /* istanbul ignore if */
     if (err) return cb(err);
 
-    var keyfile = keystore_path + path.sep + key_id;
+    var keyfile = path.join(keystore_path, key_id);
```
Comment (theopak) [inline utils/keygen.js:26]: This is one instance in the PR where `"/var" + path.sep + "logs/"` concatenation is replaced by the platform-agostic `path.join()`function.

File: lib/proxy_manager.js [hunk 2]
```diff
@@ -160,20 +160,34 @@ function loadConfigFiles(cb) {
     stop_me[proxy_config_file] = proxy;
   });
 
+  // TODO (@theopak): Consider rewriting this to make it recursive.
   fs.readdir(CONFIG_DIR, function(err, files) {
     /* istanbul ignore if */
     if (err) return cb(err);
 
+    // Require a '.json' or '.xml' file ending and reject dotfiles
+    // TODO (@theopak): Write unit tests for this.
```
Comment (theopak) [inline lib/proxy_manager.js:169]: The main change in this PR is to allow non-config files (e.g., "readme.md") in the $OAUTH_REVERSE_PROXY_CONFIG_DIR. I have yet to write unit tests although I have anecdotal evidence that the change works as expected.

File: lib/proxy_manager.js [hunk 3]
```diff
@@ -160,20 +160,34 @@ function loadConfigFiles(cb) {
...
+    var configFiles = [];
```
Comment (ryanbreen) [inline lib/proxy_manager.js]: This is a bit peevish, but I use snake_case for local variables in this project.

Comment (theopak) [inline lib/proxy_manager.js]: Ok

**Comment by theopak** [comment]
@RB when you get a chance, please let me know if this looks good to merge. I'll squash the commits now.

**Comment by ryanbreen** [comment]
Yep, I'll try to take a closer look today.  I'm not super particular about commit history, so no need to squash.

**Comment by theopak** [comment]
Already squashed everything, I don't even know if it's possible to undo a force push. Enjoy your aggressively clean commit history.

File: lib/proxy_manager.js [hunk 4]
```diff
@@ -160,20 +160,34 @@ function loadConfigFiles(cb) {
     stop_me[proxy_config_file] = proxy;
   });
 
+  // TODO (@theopak): Consider rewriting this to make it recursive.
   fs.readdir(CONFIG_DIR, function(err, files) {
     /* istanbul ignore if */
     if (err) return cb(err);
 
+    // Require a '.json' or '.xml' file ending and reject dotfiles
```
Comment (ryanbreen) [inline lib/proxy_manager.js:168]: We only support json.  Any reason to include xml config files?

File: readme.md [hunk 5]
```diff
@@ -2,15 +2,43 @@
...
+Layer to add authentication to APIs by checking caller credentials, reverse-proxying inbound traffic to your API, and then signing outbound traffic back to callers.
 
 ##### Motivation
 
-Authenticaton for web applications, particularly applications created for machine-to-machine use, is often an afterthought or implemented in an insecure or incompatible fashion.  We want a robust implementation of OAuth that can run on Windows or Unix systems in front of any HTTP-serving application and support clients written in any language.  These are two-party connections, so we can use the simplest form of OAuth: zero-legged OAuth 1.0a.
+Authentication for web applications, particularly applications created for machine-to-machine use, is often an afterthought or implemented in an insecure or incompatible fashion.  We want a robust implementation of OAuth that can run on Windows or Unix systems in front of any HTTP-serving application and support clients written in any language.  These are two-party connections, so we can use the simplest form of OAuth: zero-legged OAuth 1.0a.
```
Comment (ryanbreen) [inline readme.md:9]: oy

File: readme.md [hunk 6]
```diff
@@ -2,15 +2,43 @@
 
 [![Build Status](https://travis-ci.org/Cimpress-MCP/oauth_reverse_proxy.svg?branch=master)](https://travis-ci.org/Cimpress-MCP/oauth_reverse_proxy) [![Coverage Status](https://img.shields.io/coveralls/Cimpress-MCP/oauth_reverse_proxy.svg)](https://coveralls.io/r/Cimpress-MCP/oauth_reverse_proxy?branch=master)
 
-oauth_[|reverse_]proxy is both a proxy that can sign outbound traffic as well as a reverse proxy capable of enforcing that callers present the correct OAuth credentials.
+Layer to add authentication to APIs by checking caller credentials, reverse-proxying inbound traffic to your API, and then signing outbound traffic back to callers.
```
Comment (ryanbreen) [inline readme.md:5]: This is great!

**Comment by ryanbreen** [comment]
Looks great, thanks!

**Comment by ryanbreen** [comment]
1.5.9 published

File: lib/proxy_manager.js (hunk 4 above)
Comment (theopak) [inline lib/proxy_manager.js:168]: Oh, I don't remember what I was thinking. I'll fix this line. I'll also remove the TODO comment since it's done.

**Pull request merged**